musl/libssl1.1.x.Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM rust:alpine3.15
+
+ENV RUSTFLAGS="-C target-feature=-crt-static" \
+  PROTOC_INCLUDE="/usr/include" \
+  PROTOC="/usr/bin/protoc"
+
+RUN apk update && \
+  apk add perl musl-dev build-base bash clang git protoc protobuf protobuf-dev wget linux-headers
+
+RUN wget -c https://www.openssl.org/source/openssl-1.1.1s.tar.gz
+RUN tar -xzvf openssl-1.1.1s.tar.gz
+
+RUN cd openssl-1.1.1s && ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl shared zlib && make -j8
+
+RUN cd openssl-1.1.1s && make install
+RUN cd /etc && echo "/usr/local/ssl/lib:/lib:/usr/lib:/usr/local/lib" > ld-musl-x86_64.path
+
+ENV OPENSSL_DIR /usr/local/ssl
+ENV OPENSSL_LIB_DIR /usr/local/ssl/lib
```

Comment:
Just to make this explicit:
Why are we switching to building OpenSSL ourselves?

Reply:
We've been doing that with Debian and RHEL already, see:

https://github.com/prisma/engine-images/blob/main/rhel/libssl1.1.x.Dockerfile

Now we have full control which version we use, and we can use an old(er) Alpine base with whatever SSL we desire. This way we get a wider support of Alpines with a specific OpenSSL version.